pkg/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -224,8 +224,11 @@ func (b *builder) podSpec() (corev1.PodSpec, error) {
 		TerminationGracePeriodSeconds: address.Of(defaultTerminationGracePeriod),
 	}
 
-	if b.dk.NeedsOneAgentProbe() {
+	if b.dk.NeedsOneAgentReadinessProbe() {
 		podSpec.Containers[0].ReadinessProbe = b.getReadinessProbe()
+	}
+
+	if b.dk.NeedsOneAgentLivenessProbe() {
```

Comment:
can you please extend unit-tests to cover new condition?
```
func TestPodSpecProbes(t *testing.T) {
```

Reply:
on it :)